src/Searchbar.vue

```diff
@@ -15,41 +19,66 @@ export default {
       if (!this.data) {
         return undefined;
       }
-      const matches = [];
+      const pages = [];
       const regexes = this.value.split(' ')
         .filter(searchKeyword => searchKeyword !== '')
         .map(searchKeyword => searchKeyword.replace(/[.*+?^${}()|[\]\\]/g, '\\$&'))
-        .map(searchKeyword => new RegExp(searchKeyword, 'i'));
+        .map(searchKeyword => new RegExp(searchKeyword, 'ig'));
       this.data.forEach((entry) => {
-        const { headings, src, title } = entry;
+        const {
+          headings,
+          src,
+          title,
+          headingKeywords,
+        } = entry;
         const keywords = entry.keywords || '';
-        let searchTarget = [title].concat(keywords).concat(Object.values(headings)).join(' ');
-        let totalMatches = getTotalMatches(searchTarget, regexes);
-        if (totalMatches > 0) {
-          searchTarget = [title].concat(keywords).join(' ');
-          const isMatchingPage = getTotalMatches(searchTarget, regexes) === totalMatches;
-          if (isMatchingPage) {
-            matches.push(Object.assign(entry, { totalMatches }));
-          }
+        const fallbackTitle = title || src;
+
+        let searchTarget = [
+          fallbackTitle,
+          keywords,
+          ...Object.values(headings),
+          ...Object.values(headingKeywords),
+        ].join(' ');
+        const totalPageMatches = getTotalMatches(searchTarget, regexes);
+        if (totalPageMatches > 0) {
+          const pageHeadings = [];
           Object.entries(headings).forEach(([id, text]) => {
-            if (regexes.some(regex => regex.test(text))) {
-              searchTarget = [title].concat(keywords).concat(text).join(' ');
-              totalMatches = getTotalMatches(searchTarget, regexes);
-              matches.push({
+            const matchesHeading = regexes.some(regex => regex.test(text));
+            const matchesKeywords = headingKeywords[id] && headingKeywords[id].some(keyword =>
+              regexes.some(regex => regex.test(keyword)));
+            if (matchesHeading || matchesKeywords) {
+              searchTarget = [fallbackTitle, keywords, text, ...(headingKeywords[id] || [])].join(' ');
```

Comment:
I think it might be helpful to add some comments to explain what we're trying to do when we are changing the search target here 🙂

Reply:
Using a more specific variable name might help too!